src/main/java/edu/cornell/kfs/pmw/batch/service/impl/PaymentWorksVendorToKfsVendorDetailConversionServiceImpl.java

```diff
@@ -619,21 +609,37 @@ private String determineKfsOwnershipTypeCodeFromPmwTaxClassificationCode(Payment
         }
         return kfsOwnsershipTypeCode;
     }
-    
+
     private KfsVendorDataWrapper populateBusinessLegalName(PaymentWorksVendor pmwVendor, KfsVendorDataWrapper kfsVendorDataWrapper) {
-        kfsVendorDataWrapper.getVendorDetail().setVendorName(pmwVendor.getRequestingCompanyLegalName());
+        kfsVendorDataWrapper.getVendorDetail().setVendorName(truncateValueToMaxLength(pmwVendor.getRequestingCompanyLegalName(), VendorConstants.MAX_VENDOR_NAME_LENGTH));
         kfsVendorDataWrapper.getVendorDetail().setVendorFirstLastNameIndicator(false);
         kfsVendorDataWrapper.getVendorDetail().setVendorFirstName(KFSConstants.EMPTY_STRING);
         kfsVendorDataWrapper.getVendorDetail().setVendorLastName(KFSConstants.EMPTY_STRING);
         return kfsVendorDataWrapper;
     }
-    
+
     private KfsVendorDataWrapper populateFirstLastLegalName(PaymentWorksVendor pmwVendor, KfsVendorDataWrapper kfsVendorDataWrapper) {
         kfsVendorDataWrapper.getVendorDetail().setVendorFirstLastNameIndicator(true);
-        kfsVendorDataWrapper.getVendorDetail().setVendorFirstName(pmwVendor.getRequestingCompanyLegalFirstName());
-        kfsVendorDataWrapper.getVendorDetail().setVendorLastName(pmwVendor.getRequestingCompanyLegalLastName());
+        kfsVendorDataWrapper.getVendorDetail().setVendorLastName(truncateValueToMaxLength(pmwVendor.getRequestingCompanyLegalLastName(), VendorConstants.MAX_VENDOR_NAME_LENGTH));
+        kfsVendorDataWrapper.getVendorDetail().setVendorFirstName(truncateLegalFirstNameToMaximumAllowedLengthWhenFormattedWithLegalLastName(pmwVendor.getRequestingCompanyLegalLastName(), pmwVendor.getRequestingCompanyLegalFirstName()));
```

Comment:
Should this function take in kfsVendorDataWrapper.getVendorDetail().getVendorLastName() instead of . pmwVendor.getRequestingCompanyLegalLastName()?  I am asking because the previous line truncated the value as needed.  I am not sure if it matters really.

Reply:
Good point Jay. In this case I don't think it makes a difference because the same max vendor name length is used in both cases. However, if this was changed to use a different max length for vendor last name then it would not work correctly. I will update the code for clarity.